lib/strict-non-empty-containers/src/Data/Set/Strict/NonEmptySet.hs

```diff
@@ -0,0 +1,95 @@
+{-# LANGUAGE DeriveGeneric #-}
+{-# LANGUAGE DerivingVia #-}
+{-# LANGUAGE TupleSections #-}
+
+-- |
+-- Copyright: © 2018-2020 IOHK
+-- License: Apache-2.0
+--
+-- Provides a strict implementation of a non-empty set.
```

Comment:
What was the motivation we added this module?

Reply:
> What was the motivation we added this module?

It's used here:

```hs
data UTxOIndex = UTxOIndex
    { index
        :: !(Map AssetId (NonEmptySet TxIn))
    , balance
        :: !TokenBundle
    , utxo
        :: !(Map TxIn TxOut)
    }
    deriving (Eq, Generic, Read, Show)
```

Using a non-empty set has two advantages:

1. We don't have to manually enforce the invariant that `(asset, [])` is not allowed in the index.
2. The `Eq` instance can be auto-derived.